### PR TITLE
Adapt MediaType library change in core

### PR DESCRIPTION
**Pull request by RyanL1997**
### Description
Adapt MediaType library change in core
* Category (Enhancement, New feature, Bug fix, Test fix, Refactoring, Maintenance, Documentation)
Refactoring

### Issues Resolved
* Relate https://github.com/opensearch-project/OpenSearch/pull/8636

### Check List
- [ ] New functionality includes testing
- [ ] New functionality has been documented
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by DarshitChanpura — DISMISSED** [review]

File: src/test/java/com/amazon/dlic/auth/http/saml/HTTPSamlAuthenticatorTest.java [hunk 1]
```diff
@@ -992,7 +992,7 @@ public XContentBuilder newErrorBuilder() throws IOException {
         }
 
         @Override
-        public XContentBuilder newBuilder(XContentType xContentType, boolean useFiltering) throws IOException {
+        public XContentBuilder newBuilder(MediaType xContentType, boolean useFiltering) throws IOException {
```
Comment (willyborankin) [inline src/test/java/com/amazon/dlic/auth/http/saml/HTTPSamlAuthenticatorTest.java:995]: Same here `xContentType` vs. `mediaType`

File: src/main/java/org/opensearch/security/auditlog/impl/RequestResolver.java [hunk 2]
```diff
@@ -381,7 +382,7 @@ private static void addIndicesSourceSafe(
         final String[] indices,
         final IndexNameExpressionResolver resolver,
         final ClusterService cs,
-        final XContentType xContentType,
+        final MediaType xContentType,
```
Comment (willyborankin) [inline src/main/java/org/opensearch/security/auditlog/impl/RequestResolver.java:385]: Nit: Wdyt to rename the variable `xContentType` to `mediaType` for better readability?

File: src/test/java/com/amazon/dlic/auth/http/saml/HTTPSamlAuthenticatorTest.java [hunk 3]
```diff
@@ -1018,8 +1018,7 @@ public void sendResponse(RestResponse response) {
         }
 
         @Override
-        public XContentBuilder newBuilder(XContentType xContentType, XContentType responseContentType, boolean useFiltering)
-            throws IOException {
+        public XContentBuilder newBuilder(MediaType xContentType, MediaType responseContentType, boolean useFiltering) throws IOException {
```
Comment (willyborankin) [inline src/test/java/com/amazon/dlic/auth/http/saml/HTTPSamlAuthenticatorTest.java:1021]: Same here `xContentType` vs. `mediaType`

File: src/main/java/org/opensearch/security/auditlog/impl/RequestResolver.java (hunk 2 above)
Comment (RyanL1997) [reply src/main/java/org/opensearch/security/auditlog/impl/RequestResolver.java]: Hi @willyborankin, thats a good call for sure. I have changed them.

**Review by willyborankin — APPROVED** [review]

**Review by DarshitChanpura — APPROVED** [review]

**Review by reta — APPROVED** [review]

**Pull request merged**